### PR TITLE
DelayArchitect: init at unstable-2022-01-16

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by toastal** [comment]
correction from the docs where it's explicitly stated this needs to be the case:

> If a package is not a release but a commit from a repository, then the version part of the name _must_ be the date of that (fetched) commit. The date _must_ be in `"YYYY-MM-DD"` format. Also append `"unstable"` to the name - e.g., `"pkgname-unstable-2014-09-23"`.

File: pkgs/applications/audio/DelayArchitect/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{ lib, stdenv, fetchFromGitHub, libGL, libX11, libXext, libXrandr, libXinerama, libXcursor, freetype, alsa-lib, cmake, pkg-config, gcc-unwrapped }:
+
+stdenv.mkDerivation rec {
+  pname = "DelayArchitect";
+  version = "unstable-16-01-2022";
```
Comment (toastal) [inline pkgs/applications/audio/DelayArchitect/default.nix:5]: > If a package is not a release but a commit from a repository, then the version part of the name _must_ be the date of that (fetched) commit. The date _must_ be in `"YYYY-MM-DD"` format. Also append `"unstable"` to the name - e.g., `"pkgname-unstable-2014-09-23"`.

Should be `unstable-20222-01-16`

**Review by toastal — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/DelayArchitect/default.nix (hunk 1 above)
Comment (magnetophon) [reply pkgs/applications/audio/DelayArchitect/default.nix]: Thanks.

File: pkgs/applications/audio/DelayArchitect/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    mkdir -p $out/lib/vst3
+    cd DelayArchitect_artefacts/Release
+    cp -r VST3/Delay\ Architect.vst3 $out/lib/vst3
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/jpcima/DelayArchitect";
+    description = "A visual, musical editor for delay effects";
+    maintainers = [ maintainers.magnetophon ];
+    platforms = platforms.linux;
```
Comment (papojari) [inline pkgs/applications/audio/DelayArchitect/default.nix:37]: Why can it not be built for more platforms? The project's readme even has links to the builds for other OSs.
```suggestion
    platforms = platforms.all;
```

**Review by auroraanna — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/DelayArchitect/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{ lib, stdenv, fetchFromGitHub, libGL, libX11, libXext, libXrandr, libXinerama, libXcursor, freetype, alsa-lib, cmake, pkg-config, gcc-unwrapped }:
+
+stdenv.mkDerivation rec {
+  pname = "DelayArchitect";
```
Comment (papojari) [inline pkgs/applications/audio/DelayArchitect/default.nix:4]:
```suggestion
  pname = "delayarchitect";
```
[Can't have uppercase letters](https://nixos.org/manual/nixpkgs/stable/#sec-package-naming).

**Review by auroraanna — CHANGES_REQUESTED** [review]

**Comment by papojari** [comment]
```
[nix-shell:~/.cache/nixpkgs-review/pr-162426/results/DelayArchitect/lib/vst3/Delay Architect.vst3/Contents/x86_64-linux]$ steam-run Delay\ Architect.so 
/nix/store/yq37f80a8vb1cp2agdg07a66gm75fvk8-steam-run: line 15: exec: Delay Architect.so: not found
```
Not sure how to run it.

File: pkgs/applications/audio/DelayArchitect/default.nix (hunk 2 above)
Comment (magnetophon) [reply pkgs/applications/audio/DelayArchitect/default.nix]: Thanks for the feedback, all done.

**Comment by magnetophon** [comment]
> Not sure how to run it.

It's a plugin. You need a host that can run VST3's, like Ardour, for example.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -1724,6 +1724,8 @@ with pkgs;
 
   deltachat-cursed = callPackage ../applications/networking/instant-messengers/deltachat-cursed { };
 
+  DelayArchitect = callPackage ../applications/audio/DelayArchitect { };
```
Comment (papojari) [inline pkgs/top-level/all-packages.nix:1727]:
```suggestion
  delayarchitect = callPackage ../applications/audio/delayarchitect { };
```
I forgot to tell you but this also needs to be lowercase. Not sure if it's a good idea to have the result named `DelayArchitect`, with uppercase letters, either.

**Review by auroraanna — CHANGES_REQUESTED** [review]

**Comment by papojari** [comment]
Result of `nixpkgs-review pr 162426` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>delayarchitect</li>
  </ul>
</details>

**Comment by papojari** [comment]
There was an error though:
```
error: build log of '/nix/store/4nm6az3bfm8jgny139pcf20yygdnhs91-delayarchitect-unstable-2022-01-16.drv' is not available
```
but the result didn't fail. :shrug:

**Review by auroraanna — APPROVED** [review]

**Pull request merged**